### PR TITLE
Add support for underscores in OCaml type expressions e.g. '_ list'.

### PR DESCRIPTION
**Pull request by mjambon**
It's a nice little feature that I use to it would be great if semgrep supported it.

### CODE REVIEW

**Comment by mjambon** [comment]
Semgrep fails because it's not using this branch of pfff.

**Comment by mjambon** [comment]
I excluded the folder containing the ocaml test cases so that semgrep doesn't choke on them.

File: h_program-lang/AST_generic.ml [hunk 1]
```diff
@@ -947,6 +947,7 @@ and type_ =
   | TyNameApply of name * type_arguments
 
   | TyVar of ident (* type variable in polymorphic types (not a typedef) *)
+  | TyAny of ident (* anonymous type, '_' in OCaml *)
```
Comment (aryx) [inline h_program-lang/AST_generic.ml:950]: could be TyAny of tok then, if you know it's always _, there is no need for the ident value.

File: lang_ml/parsing/ast_ml.ml [hunk 2]
```diff
@@ -59,6 +59,7 @@ type todo_category = string wrap
 type type_ =
   | TyName of name (* include builtins *)
   | TyVar of ident (* 'a *)
+  | TyAny of ident (* _ *)
```
Comment (aryx) [inline lang_ml/parsing/ast_ml.ml:62]: same here, TyAny of tok

**Review by aryx — APPROVED** [review]
Great! This did not add any conflicts in the grammar right?
menhir didn't warn about additional warnings?
(would be great to ensure this in dune, that we don't add any more conflicts
than there was before; we had this before when we were using Makefile but we didn't
port this check to dune).

File: h_program-lang/AST_generic.ml (hunk 1 above)
Comment (mjambon) [reply h_program-lang/AST_generic.ml:950]: I thought we might find the same feature in another language, where the symbol is not `_`.

File: lang_ml/parsing/ast_ml.ml (hunk 2 above)
Comment (mjambon) [reply lang_ml/parsing/ast_ml.ml:62]: This was a bit lazy on my part. It would be clearer to only have a tok as argument.

**Comment by mjambon** [comment]
I didn't see new conflicts reported by menhir.

**Pull request merged**